x-pack/plugins/apm/server/lib/apm_telemetry/apm_telemetry.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+
+// Support telemetry for additional agent types by appending definitions in
+// mappings.json and the AgentName enum.
+
+export enum AgentName {
+  Python = 'python',
+  Java = 'java',
+  NodeJs = 'nodejs',
+  JsBase = 'js-base',
+  Ruby = 'ruby',
+  GoLang = 'go'
+}
+
+export interface ApmTelemetry {
+  has_any_services: boolean;
+  services_per_agent: { [agentName in AgentName]: number };
+}
+
+export const APM_TELEMETRY_DOC_ID = 'apm-telemetry';
+
+export function createApmTelementry(
+  agentNames: AgentName[] = []
+): ApmTelemetry {
+  return agentNames.reduce(
+    (apmTelemetry, agentName) => {
+      if (agentName && Object.values(AgentName).includes(agentName)) {
+        apmTelemetry.has_any_services = true;
+        apmTelemetry.services_per_agent[agentName]++;
+      }
+      return apmTelemetry;
+    },
+    {
+      has_any_services: false,
+      services_per_agent: Object.values(AgentName).reduce(
+        (memo, agentName: AgentName) => Object.assign(memo, { [agentName]: 0 }),
+        {}
+      )
+    } as ApmTelemetry
+  );
+}
```

Comment:
What do you think about something like this?

```js
  const validAgentNames = agentNames.filter(
    agentName => Object.values(AgentName).includes(agentName)
  );
  return {
    has_any_services: validAgentNames.length > 0,
    services_per_agent: countBy(validAgentNames)
  };
```

It's not 100% compatible with yours, since it omits the zero values.